routes/truckRoutes.js

```diff
@@ -1 +1,28 @@
-// put your Express router code in here
\ No newline at end of file
+var express = require('express');
+var trucks = require('../trucks');
+var bodyParser = require('body-parser');
+var router = express.Router();
+var urlEncoded = bodyParser.urlencoded({ extended: false });
+
+router.route('/')
+  .get(function (request, response){
+    var truckList = trucks.getTrucks();
+  
+    response.send(truckList);
+})
+.post(urlEncoded, function(request, response){
+  var newTruck = trucks.addTruck(truck); 
+  response.send(newTruck);
+})
+
+router.route('/:name') 
+  .get(function(request, response){
+  var truck = trucks.getTruck(request.params.name);
+  response.send(truck);
+})
+
+.delete(function(request, response){
+  var name = trucks.removeTruck(request.params.name);
```

Comment:
Calling removeTruck is the correct thing to do, but your route still needs to send a response back to the caller, or it will just hang here in the request cycle. In this case, you don't need removeTruck to return the name, and you don't need to return the name to the caller. You do need to set a status and send the response, however.